### PR TITLE
Refactor commit_message to allow custom prompts

### PR DESCRIPTION
**Pull request by Jasha10**
Hello,

Thank you for open sourcing this project!

This PR allows the user to pass a custom `--prompt` via the command line.

### CODE REVIEW

File: gpt_commit_msg.py [hunk 1]
```diff
@@ -114,9 +111,17 @@ def main():
...
         args.model = "gpt-3.5-turbo"
 
+    if args.prompt:
+        prompt = args.prompt
+    else:
+        prompt = """Write a git commit message for the following. The message
+                starts with a one-line summary of 60 characters, followed by a
+                blank line, followed by a longer but concise description of the
+                change."""
+
```
Comment (rtwfroody) [inline gpt_commit_msg.py]: Instead of doing this, can you pass the default prompt using `default=` as a parameter to `--prompt` argument?

**Review by rtwfroody — CHANGES_REQUESTED** [review]
Thanks for sharing this. Looks like a nice improvement.

File: gpt_commit_msg.py (hunk 1 above)
Comment (Jasha10) [reply gpt_commit_msg.py]: Sure, I've updated the code in commit https://github.com/rtwfroody/gpt-commit-msg/pull/1/commits/79938e38d8532051bc4be6ee4dc646ed0f4cb67e.
Thanks for the feedback!

**Comment by Jasha10** [comment]
FYI The newline characters and spaces in the prompt are tokenized and sent to GPT:
```python
                        default="""Write a git commit message for the following. The message
                                starts with a one-line summary of 60 characters, followed by a
                                blank line, followed by a longer but concise description of the
                                change."""
```
This default prompt has a newline character and a bunch of spaces at the beginning of each line (except for the first line). If this is of concern to you @rtwfroody, I can submit another PR to fix it.

**Review by rtwfroody — APPROVED** [review]
Thank you!

**Comment by rtwfroody** [comment]
Good point about the tokens. I kind of assumed that any amount of whitespace is just a single token, but it sounds like that's not the case. That would be a nice thing to fix as well.

**Comment by Jasha10** [comment]
> I kind of assumed that any amount of whitespace is just a single token

Using the [openai/tiktoken](https://github.com/openai/tiktoken) library (which is a dependency of `gpt-commit-msg`) I've been able to confirm that different amounts of whitespace result in different tokens, and newlines are treated differently from spaces.

I'll follow up with another PR.

**Pull request merged**